invenio_communities/communities/services/components.py

```diff
@@ -271,13 +271,24 @@ def unmark(self, identity, data=None, record=None, **kwargs):
 class CommunityThemeComponent(ServiceComponent):
     """Service component for Community PIDs."""
 
-    def set_theme(self, identity, data=None, record=None, errors=None, **kwargs):
+    def update(self, identity, data=None, record=None, errors=None, **kwargs):
         """Inject parsed theme to the record."""
-        record["theme"] = data
 
-    def delete_theme(self, identity, data=None, record=None, errors=None, **kwargs):
-        """Delete theme from the record."""
-        record.pop("theme", None)
+        if "theme" in data:
+            # if theme set to None then it is a delete operation
+            if data["theme"] is None:
+                self.service.require_permission(identity, "delete_theme", record=record)
+                # delete theme from record and data
+                record.pop("theme", None)
+                data.pop("theme", None)
+            else:
+                # check update permissions for theme
+                self.service.require_permission(identity, "set_theme", record=record)
+                record["theme"] = data["theme"]
+        else:
```

Comment:
this else will never be reached when we use the resource (I know that it will come in later stages) since theme is always in the payload (we use PUT not PATCH)

Reply:
You are right! Wrong else placement....